sklearn/exceptions.py

```diff
@@ -154,3 +155,10 @@ class UndefinedMetricWarning(UserWarning):
     .. versionchanged:: 0.18
        Moved from sklearn.base.
     """
+
+
+class KernelWarning(UserWarning):
+    """Custom warning to capture kernel issues
```

Comment:
It's not clear what a kernel is in this context

Reply:
Yes, I'll use `SPDSpectrumWarning`.